### PR TITLE
TASK-5

### PR DESCRIPTION
**Pull request by LoraBard**
**Link to app**: [https://dxxidkueuub78.cloudfront.net/](https://dxxidkueuub78.cloudfront.net/)
**Link to FE PR**: [https://github.com/LoraBard/nodejs-aws-fe/pull/3](https://github.com/LoraBard/nodejs-aws-fe/pull/3)

1 - File serverless.yml contains configuration for importProductsFile function
3 - The importProductsFile lambda function returns a correct response which can be used to upload a file into the S3 bucket
4 - Frontend application is integrated with importProductsFile lambda
5 - The importFileParser lambda function is implemented and serverless.yml contains configuration for the lambda

**Additional tasks**:
- async/await is used in lambda functions
- importProductsFile lambda is covered by unit tests
- At the end of the stream the lambda function should move the file from the uploaded folder into the parsed folder (move the file means that file should be copied into parsed folder, and then deleted from uploaded folder)

### CODE REVIEW

File: shop-service/import-service/handlers/importFileParser.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+export const importFileParser = async (event: S3Event, context: Context): Promise<any> => {
+  console.log(
+    `Event:\n ${JSON.stringify(event)}\nContext:\n${JSON.stringify(context)}`
+  );
+  try {
+    const { BUCKET } = process.env;
+    const s3 = new AWS.S3({ region: "eu-west-1" });
+
+    await new Promise((resolve, reject) => {
+      event.Records.forEach((record) => {
```
Comment (inshatan) [inline shop-service/import-service/handlers/importFileParser.ts]: In the case of several files are uploaded at once, you will resolve the Promise after the first processed file.
From the async/await point of view, you should create a Promise for each record and wait for all of them are resolved.

**Comment by inshatan** [comment]
An error occurred while trying to upload csv-file through FE Application

Capture

Try to add  signatureVersion: 'v4' to the options of the AWS.S3 constructor

File: shop-service/import-service/handlers/importFileParser.ts (hunk 1 above)
Comment (LoraBard) [reply shop-service/import-service/handlers/importFileParser.ts]: thanks for your comment! Fixed it

File: shop-service/import-service/handlers/importFileParser.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  console.log(
+    `Event:\n ${JSON.stringify(event)}\nContext:\n${JSON.stringify(context)}`
+  );
+  try {
+    const { BUCKET } = process.env;
+    const s3 = new AWS.S3({ region: "eu-west-1", signatureVersion: "v4" });
+
+    await Promise.all(
+      event.Records.map(async (record) => {
+        return await new Promise((resolve, reject) => {
```
Comment (maximsan) [inline shop-service/import-service/handlers/importFileParser.ts:24]: This await looks redundant

**Review by maximsan — APPROVED** [review]
Looks fine for me

**Pull request merged**